libs/core-ui/src/lib/util/getDataBalanceChartOptions.ts

```diff
@@ -0,0 +1,287 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import {
+  ColorAxisOptions,
+  Point,
+  SeriesOptionsType,
+  TitleOptions
+} from "highcharts";
+
+import { IHighchartsConfig } from "../Highchart/IHighchartsConfig";
+import {
+  ApprovedAggregateMeasures,
+  ApprovedDistributionMeasures,
+  ApprovedFeatureBalanceMeasures,
+  getAggregateBalanceMeasures,
+  getDistributionBalanceMeasures,
+  getFeatureBalanceMeasures,
+  IAggregateBalanceMeasures,
+  IDistributionBalanceMeasures,
+  IFeatureBalanceMeasures
+} from "../Interfaces/DataBalanceInterfaces";
+
+export function getFeatureBalanceChartOptions(
+  featureBalanceMeasures: IFeatureBalanceMeasures,
+  selectedFeature: string,
+  selectedMeasure: string,
+  datasetName?: string
+): IHighchartsConfig {
+  const measure = ApprovedFeatureBalanceMeasures.get(selectedMeasure);
+  const uniqueClasses = featureBalanceMeasures.uniqueClasses[selectedFeature];
+  if (!measure || !uniqueClasses) {
+    return {};
+  }
+
+  const data: number[][] = [];
+  uniqueClasses.forEach((classA, colIndex) => {
+    uniqueClasses.forEach((classB, rowIndex) => {
+      const featureValue = getFeatureBalanceMeasures(
+        featureBalanceMeasures.measures,
+        selectedFeature,
+        classA,
+        classB
+      )[measure.varName];
+
+      // Feature values don't exist for the diagonal of the heatmap (i.e. the same class)
+      // Additionally, some feature values may be undefined/invalid/NaN
+      if (featureValue && colIndex !== rowIndex) {
+        // Add the feature value for comparing class A (col index) to class B (row index)
+        // The feature value may not be the same as the feature value when comparing class B to class A (swapped row and col)
+        data.push([colIndex, rowIndex, featureValue]);
+      }
+    });
+  });
+
+  const titleOptions = {
+    text: `Comparing ${selectedMeasure} on all classes of ${selectedFeature}`
+  } as TitleOptions;
+  if (datasetName) {
+    titleOptions.text += ` in ${datasetName}`;
+  }
+
+  const colorAxisOptions = {} as ColorAxisOptions;
+  if (measure.range) {
+    [colorAxisOptions.min, colorAxisOptions.max] = measure.range;
+  } else {
+    const featureValues = data.map((row) => row[2]);
```

Comment:
I wonder if a reduce on data which calculates the min/max in a tuple might be slightly cleaner and more performant since it would loop just once instead of three times over the data.  But maybe this is written in a clearer way anyway.  Not sure.